internal/lsp/location_links.go

```diff
@@ -31,6 +31,7 @@ func refTargetToLocationLink(target *decoder.ReferenceTarget) lsp.LocationLink {
 		OriginSelectionRange: HCLRangeToLSP(target.OriginRange),
 		TargetURI:            lsp.DocumentURI(targetUri),
 		TargetRange:          HCLRangeToLSP(target.Range),
+		TargetSelectionRange: HCLRangeToLSP(target.Range),
 	}
 
 	if target.DefRangePtr != nil {
```

Comment:
Do we want to add a test case for the situation when `TargetSelectionRange` gets overwritten by this condition?

Reply:
Added a test case.